app/src/docs/components/Search/index.jsx

```diff
@@ -0,0 +1,114 @@
+    }
+
+    const closeOverlay = () => {
+        setOverlayVisible(false)
+        toggleOverlay()
+    }
+
+    const onKeyDown = useCallback((event) => {
+        if (event.key === 'Escape') {
+            setOverlayVisible(false)
+        }
+    }, [])
+
+    useEffect(() => {
+        window.addEventListener('keydown', onKeyDown)
+        return () => {
+            window.removeEventListener('keydown', onKeyDown)
+        }
+    }, [onKeyDown])
+
+    return (overlayVisible
+        ? (
+            <div
+                className={styles.searchOverlay}
+            >
+                <BodyClass className={overlayVisible && 'overflow-hidden'} />
+                <div className={styles.searchContainer}>
+                    <SearchInput
+                        value={query}
+                        onChange={onSearchChange}
+                        placeholder="Search"
+                        className={styles.docsSearchInput}
+                        hideClearButton
+                        autoFocus
+                    />
+                    <button
+                        type="button"
+                        className={styles.exitButton}
+                        onClick={closeOverlay}
+                    >
+                        <SvgIcon name="cross" />
+                    </button>
+                </div>
+                <div className={styles.searchResults}>
+                    <ul>
+                        {searchResults.map((result, resultIndex) => (
+                            resultIndex <= 10
+                                ? (
+                                    <li key={result.id}>
+                                        <Link onClick={resultClick} className={styles.resultHeading} to={result.id}>
+                                            {result.title}
+                                        </Link>
+                                        <RawHtml className={styles.searchResultSnippet}>
+                                            {result.textSnippet}
+                                        </RawHtml>
+                                        <span className={styles.resultSection}>
+                                            {result.section}
+                                        </span>
+                                    </li>)
+                                : null
+                        ))}
+                        {!searchResults.length && !!query.length && (
+                            <React.Fragment>
+                                <p className={styles.noResults}>No results found for <strong>{query}</strong></p>
+                                <p className={styles.noResultsMoreInfo}>
+                                    <Translate value="docs.search.noResults" tag="span" />
+                                    <br className={styles.mobileOnlyBreak} />
+                                    <a rel="noopener noreferrer" target="_blank" href={links.community.devForum}> Community Dev Forum </a>
```

Comment:
It is a waste of time, but I just can't see a way out it right now.

Reply:
We actually could use markdown + i18n params. Throwing in HTML tags into `msgstr` feels like… a surrender. 😅

Example of how it could work:

```po
msgid "docs##search##noResultsFoundFor"
msgstr "No results found for **${query}**"

msgid "docs##search##noResults"
msgstr "Please try a different search or ask on our [Community Dev Forum](%{url}) instead."
```

Then

```jsx
<Translate key="docs.search.noResultsFoundFor" query="clogger" />
/* => <span>No results found for <strong>clogger</strong></span>*/

// etc.
```

Thoughts?

Reply:
Thanks @mondoreale, fixed - it was just a bit tricky to get the media query I needed to perform in the translation snippet but I got there : )